src/components/UploadingJobsDisplay/utils/upload.js

```diff
@@ -91,33 +82,38 @@ export const uploadFile = (file, url, headers, onprogress) => {
         }
       };
 
-      xhr.send(octet);
+      xhr.send(file);
     } catch (e) {
-      onFileReadError(e);
+      onFileUploadError(e);
     }
   });
 };
 
 export const uploadFiles = (
   files,
+  filesMeta,
   url,
   headers,
   onXHRprogress,
   onXHRload,
   onXHRerror,
 ) => {
-  const filesArr = Object.values(files);
   let promise = Promise.resolve();
 
-  filesArr.forEach(file => {
-    const queryParams = `?fileId=${file.id}&uploadDefinitionId=${file.uploadDefinitionId}`;
-    const urlWithQueryParams = url + queryParams;
-
-    promise = promise
-      .then(() => uploadFile(file, urlWithQueryParams, headers, onXHRprogress))
-      .then(onXHRload)
-      .catch(onXHRerror);
-  });
+  Object
+    .keys(filesMeta)
```

Comment:
Is there a reason to use `Object.keys()`? `Object.values()` seems more appropriate.

Reply:
I would leave `keys` approach for consistency as everywhere I use this approach. That includes rendering the files and other operations like this one.